server/models/estimationTaskModel.js

```diff
@@ -383,5 +383,35 @@ estimationTaskSchema.statics.moveTaskOutOfFeatureByEstimator = async (featureInp
     return await task.save();
 }
 
+
+
+estimationTaskSchema.statics.requestEditPermissionOfTaskByEstimator = async (taskInput, estimator) => {
+    if (!estimator || !userHasRole(estimator, SC.ROLE_ESTIMATOR))
+        throw new AppError('Not an estimator', EC.INVALID_USER, EC.HTTP_BAD_REQUEST)
+
+    let task = await EstimationTaskModel.findById(taskInput.task_id)
+    if (!task)
+        throw new AppError('Task not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    let estimation = await EstimationModel.findOne({"_id": task.estimation._id})
+    if (!estimation)
+        throw new AppError('Estimation not found', EC.NOT_FOUND, EC.HTTP_BAD_REQUEST)
+
+    if (!_.includes([SC.STATUS_ESTIMATION_REQUESTED, SC.STATUS_CHANGE_REQUESTED], estimation.status))
+        throw new AppError("Estimation has status as [" + estimation.status + "]. Estimator can only request edit task into those estimations where status is in [" + SC.STATUS_ESTIMATION_REQUESTED + ", " + SC.STATUS_CHANGE_REQUESTED + "]", EC.INVALID_OPERATION, EC.HTTP_BAD_REQUEST)
+
+    if (!estimation.estimator._id == estimator._id)
+        throw new AppError('Not an estimator', EC.INVALID_USER, EC.HTTP_BAD_REQUEST)
+    let changeRequested = false
+    if(task.estimator.changeRequested)
+        changeRequested = false
+    else
+        changeRequested = true
+    task.estimator.changeRequested = changeRequested
```

Comment:
Can write above as

task.estimator.changeRequested = !task.estimator.changeRequested

Reply:
Also please also add task.estimator.changedInThisIteration to true (so that we can show this task has changed in this iteration)